account_asset_management/models/account_invoice.py

```diff
@@ -0,0 +1,147 @@
+                        analytic_line = analytic_line_tuple[2]
+                        analytic_line['amount'] = round(
+                            analytic_line['amount'] / line_qty,
+                            dp.precision_get('Account'))
+                        analytic_line['unit_amount'] = round(
+                            analytic_line['unit_amount'] / line_qty, 2)
+                    line_to_create = line_qty
+                    while line_to_create > 1:
+                        line_to_create -= 1
+                        new_line = copy.deepcopy(line_tuple)
+                        new_lines.append(new_line)
+                    # Compute rounding difference and apply it on the first
+                    # line
+                    line['quantity'] += round(
+                        origin_line['quantity'] - line['quantity'] * line_qty,
+                        2)
+                    line['debit'] += round(
+                        origin_line['debit'] - line['debit'] * line_qty,
+                        dp.precision_get('Account'))
+                    line['credit'] += round(
+                        origin_line['credit'] - line['credit'] * line_qty,
+                        dp.precision_get('Account'))
+                    i = 0
+                    for analytic_line_tuple in line['analytic_line_ids']:
+                        analytic_line = analytic_line_tuple[2]
+                        origin_analytic_line = \
+                            origin_line['analytic_line_ids'][i][2]
+                        analytic_line['amount'] += round(
+                            origin_analytic_line['amount'] - analytic_line[
+                                'amount'] * line_qty,
+                            dp.precision_get('Account'))
+                        analytic_line['unit_amount'] += round(
+                            origin_analytic_line['unit_amount'] -
+                            analytic_line[
+                                'unit_amount'] * line_qty,
+                            dp.precision_get('Account'))
+                        i += 1
+        move_lines.extend(new_lines)
+        return move_lines
+
+    @api.multi
+    def action_move_create(self):
+        res = super(AccountInvoice, self).action_move_create()
+        for inv in self:
+            move = inv.move_id
+            assets = [aml.asset_id for aml in
+                      [x for x in move.line_ids if x.asset_id]]
+            for asset in assets:
+                asset.code = inv.move_name
+                asset_line_name = asset._get_depreciation_entry_name(0)
+                asset.depreciation_line_ids[0].with_context(
+                    {'allow_asset_line_update': True}
+                ).name = asset_line_name
+        return res
+
+    @api.multi
+    def action_cancel(self):
+        assets = self.env['account.asset']
+        for inv in self:
+            move = inv.move_id
+            assets = move.line_ids.mapped('asset_id')
```

Comment:
so if you are cancelling more than one invoice your actually deleting only the last ones.
As this operation should not run very often on many records you could do:
```python
def action_cancel(self):
    super().action_cancel()
    self.mapped('move_id.line_ids.asset_id').unlink()
```

Reply:
@luc-demeyer Would be great to have your input on this, as I don't know if it's intended to delete only the last ones....